src/Illuminate/Cache/RedisTagSet.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Illuminate\Cache;
+
+use Illuminate\Support\LazyCollection;
+
+class RedisTagSet extends TagSet
+{
+    /**
+     * Add a reference entry to the tag set's underlying sorted set.
+     *
+     * @param  string  $key
+     * @param  int  $ttl
+     * @return void
+     */
+    public function addEntry(string $key, int $ttl = 0)
+    {
+        $ttl = $ttl > 0 ? now()->addSeconds($ttl)->getTimestamp() : -1;
+
+        foreach ($this->tagIds() as $tagKey) {
+            $this->store->connection()->zadd($this->store->getPrefix().$tagKey, $ttl, $key);
+        }
+    }
+
+    /**
+     * Get all of the cache entry keys for the tag set.
+     *
+     * @return \Illuminate\Support\LazyCollection
+     */
+    public function entries()
+    {
+        return LazyCollection::make(function () {
+            foreach ($this->tagIds() as $tagKey) {
+                $cursor = $defaultCursorValue = '0';
+
+                do {
+                    [$cursor, $entries] = $this->store->connection()->zscan(
+                        $this->store->getPrefix().$tagKey,
+                        $cursor,
+                        ['match' => '*', 'count' => 1000]
+                    );
+
+                    if (! is_array($entries)) {
+                        break;
+                    }
+
+                    $entries = array_unique(array_keys($entries));
+
+                    if (count($entries) === 0) {
+                        continue;
+                    }
+
+                    foreach ($entries as $entry) {
+                        yield $entry;
+                    }
+                } while (((string) $cursor) !== $defaultCursorValue);
+            }
+        });
+    }
+
+    /**
+     * Remove the stale entries from the tag set.
+     *
+     * @return void
+     */
+    public function flushStaleEntries()
+    {
+        $this->store->connection()->pipeline(function ($pipe) {
+            foreach ($this->tagIds() as $tagKey) {
+                $pipe->zremrangebyscore($this->store->getPrefix().$tagKey, 0, now()->getTimestamp());
+            }
+        });
+    }
+
+    /**
+     * Flush the tag from the cache.
+     *
+     * @param  string  $name
+     */
+    public function flushTag($name)
+    {
+        return $this->resetTag($name);
+    }
+
+    /**
+     * Reset the tag and return the new tag identifier.
+     *
+     * @param  string  $name
+     * @return string
+     */
+    public function resetTag($name)
+    {
+        $this->store->forget($this->tagKey($name));
+
+        return $this->tagId($name);
+    }
+
+    /**
+     * Get the unique tag identifier for a given tag.
+     *
+     * @param  string  $name
+     * @return string
+     */
+    public function tagId($name)
+    {
+        return "tag:{$name}:entries";
+    }
+
+    /**
+     * Get the tag identifier key for a given tag.
+     *
+     * @param  string  $name
+     * @return string
+     */
+    public function tagKey($name)
```

Comment:
`tagKey` used to return just "tag:$name:key" whose value pointed to the name of tag reference key, while `tagId` returned the reference key itself (`laravel:6307213384953462982890:standard_ref`). Technically, if somebody used those for whatever reason, it will be a breaking change for them.

Same goes for `resetTag` - if somebody calls that and expects the tag key to be present in cache afterwards (which is what used to happen), this is also technically a breaking change.

Both of these aren't major at all and I doubt many people used these at all, but it's worth mentioning.